id: 7857962b4d45e602b8ad
bug_link: https://syzkaller.appspot.com/bug?extid=7857962b4d45e602b8ad
title: general protection fault in __dentry_path
source_page: https://syzkaller.appspot.com/upstream/fixed
trigger_commit: 0a1d52994d440e21def1c2174932410b4f2a98a1
fix_commit: 8ed0579c12b2fe56a1fac2f712f58fc26c1dc49b
datetime: '2019-02-28T08:57:32-08:00'
fix_commit_message: 'kvm: properly check debugfs dentry before using it


  debugfs can now report an error code if something went wrong instead of

  just NULL.  So if the return value is to be used as a "real" dentry, it

  needs to be checked if it is an error before dereferencing it.


  This is now happening because of ff9fb72bc077 ("debugfs: return error

  values, not NULL").  syzbot has found a way to trigger multiple debugfs

  files attempting to be created, which fails, and then the error code

  gets passed to dentry_path_raw() which obviously does not like it.


  Reported-by: Eric Biggers <ebiggers@kernel.org>

  Reported-and-tested-by: syzbot+7857962b4d45e602b8ad@syzkaller.appspotmail.com

  Cc: "Radim Krčmář" <rkrcmar@redhat.com>

  Cc: kvm@vger.kernel.org

  Acked-by: Paolo Bonzini <pbonzini@redhat.com>

  Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

  Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

  '
submodule:
- virt/kvm
hunk_count: 1
covered_count: 1
